src/Tester/UnitTestUtils.cs

```diff
@@ -76,5 +75,28 @@ public static void ConfigureThreadPoolSettingsForStorageTests(int NumDotNetPoolT
             ServicePointManager.DefaultConnectionLimit = NumDotNetPoolThreads; // 1000;
             ServicePointManager.UseNagleAlgorithm = false;
         }
+
+        public static async Task WithTimeout(this Task taskToComplete, TimeSpan timeout, string message)
```

Comment:
This was copied from TaskExtensions. Exactly the same code. TaskExtensions is internal. We could make TaskExtensions public.